internals-js/src/definitions.ts

```diff
@@ -2058,7 +2072,7 @@ export class FieldDefinition<TParent extends CompositeType> extends NamedSchemaE
   }
 
   get coordinate(): string {
-    const parent = this.parent;
+    const parent = this._parent; // TODO: @pcmanus does this change look right to you?
```

Comment:
Yes, looks ok to me (and fwiw, the commit I link above removes the TODO, assuming you're ok with that earlier suggestion). To expand, I can't think of code calling this method on a detached element so using `this.parent` would probably not break anything, nor can I think of a definitive reason why one option would be strictly better than the other: we use this in a number of error message so there is probably cases where actually getting an actual error clearly stating the element is detached help debugging more than throwing while generating the error message, but I can imagine other cases where throwing would be better. Overall, happy either way but a slight preference for using `this._parent` as you did.